xbmc/pvr/channels/PVRChannelGroup.cpp

```diff
@@ -81,10 +81,24 @@ void CPVRChannelGroup::OnInit()
   CServiceBroker::GetSettingsComponent()->GetSettings()->RegisterCallback(this, {
     CSettings::SETTING_PVRMANAGER_BACKENDCHANNELORDER,
     CSettings::SETTING_PVRMANAGER_USEBACKENDCHANNELNUMBERS,
+    CSettings::SETTING_PVRMANAGER_USEBACKENDCHANNELNUMBERSALWAYS,
     CSettings::SETTING_PVRMANAGER_STARTGROUPCHANNELNUMBERSFROMONE
   });
 }
 
+namespace
+{
+
+bool UsingBackendChannelNumbers(const std::shared_ptr<CSettings>& settings)
+{
+  return settings->GetBool(CSettings::SETTING_PVRMANAGER_USEBACKENDCHANNELNUMBERS) &&
+         (CServiceBroker::GetPVRManager().Clients()->EnabledClientAmount() == 1 ||
+          (settings->GetBool(CSettings::SETTING_PVRMANAGER_USEBACKENDCHANNELNUMBERSALWAYS) &&
+           CServiceBroker::GetPVRManager().Clients()->EnabledClientAmount() > 1));
```

Comment:
would it be worth saving `.Clients()->EnabledClientAmount()` to a `const int` before?
don't expect a huge speedup (if any at all) but those CB-calls are somehow expensive, aren't they?